<launch>
  <group ns="quadrotor">
    <node pkg="quadrotor_simulator"
      type="quadrotor_simulator_so3"
      name="quadrotor_simulator_so3"
      output="screen">
      <param name="rate/odom" value="100.0"/>
      <remap from="~odom" to="odom"/>
      <remap from="~cmd" to="so3_cmd"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      name="standalone_nodelet"
      args="manager"
      output="screen"/>

    <node pkg="nodelet"
      type="nodelet"
      name="controllers_manager"
      args="load controllers_manager/ControllersManager standalone_nodelet"
      output="screen">
      <rosparam file="$(find controllers_manager)/config/controllers.yaml"/>
      <rosparam file="$(find so3_control)/config/gains.yaml"/>
      <param name="line_tracker/default_v_des" value="2.0"/>
      <param name="line_tracker/default_a_des" value="1.0"/>
      <remap from="~odom" to="odom"/>
      <remap from="~cmd" to="position_cmd"/>
      <remap from="~path/trajectory_tracker/path" to="path"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      args="load so3_control/SO3ControlNodelet standalone_nodelet"
      name="so3_control"
      required="true"
      output="screen">
      <rosparam file="$(find so3_control)/config/gains.yaml"/>
      <param name="mass" value="0.5"/>
      <remap from="~odom" to="odom"/>
      <remap from="~position_cmd" to="position_cmd"/>
      <remap from="~so3_cmd" to="so3_cmd"/>
      <remap from="~motors" to="motors"/>
    </node>

    <node pkg="trajectory_tracker"
      type="sample_planner"
      name="planner"
      output="screen">
      <!-- <remap from="~path" to="path"/> -->
      <!-- <remap from="~goal" to="goal"/> -->
     <!-- <remap from="~transition" to="controllers_manager/transition"/> -->
    </node>

    <node pkg="mesh_visualization"
      type="mesh_visualization"
      name="mesh_visualization"
      output="screen">
      <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
      <param name="color/r" value="0.0"/>
      <param name="color/g" value="0.0"/>
      <param name="color/b" value="1.0"/>
      <param name="color/a" value="0.7"/>
      <remap from="~odom" to="odom"/>
    </node>

    <node pkg="rviz"
      type="rviz"
      name="rviz"
      args="-d $(find quadrotor_simulator)/launch/rviz_config.rviz"/>
  </group>
</launch>
